src/main/java/org/opentripplanner/client/parameters/TripPlanParameters.java

```diff
@@ -57,6 +74,61 @@ public boolean isArriveBy() {
     }
   }
 
+  public enum OptimizeType {
+    QUICK,
+    SAFE,
+    FLAT,
+    GREENWAYS,
+    TRIANGLE,
+  }
+
+  public static class InputBanned {
```

Comment:
Do you think, it makes sense to use the builder pattern in this class, as with TripPlanParameters.
Because using the constructor in Java could be a bit error-prone if you only want to filter by one criterion but have to pass five parameters?

Reply:
Yes, please use a builder. It makes the library much easier to use.

Intellij even has an automatic conversion tool.

Since this will make the class quite large, please use move it into its own file.

Reply:
I changed the banned and triangle parameters as Builder classes.